pkg/uiserver/uiserver.go

```diff
@@ -53,6 +61,63 @@ func RewriteAssets(fs http.FileSystem, cfg *config.Config, updater UpdateContent
 
 	rewrite("/index.html")
 	rewrite("/diagnoseReport.html")
+
+	overrideDistroAssetsRes(fs, cfg, updater)
+}
+
+func overrideDistroAssetsRes(fs http.FileSystem, cfg *config.Config, updater UpdateContentFunc) {
+	exePath, err := os.Executable()
+	if err != nil {
+		log.Fatal("Failed to get work dir", zap.Error(err))
+	}
+
+	distroResDir := path.Join(path.Dir(exePath), distro.DistroResFolderName)
+	info, err := os.Stat(distroResDir)
+	if err != nil || !info.IsDir() {
+		// just ignore
+		return
+	}
+
+	override := func(assetName string) {
+		assetPath := path.Join("/", distro.DistroResFolderName, assetName)
+		targetFile, err := fs.Open(assetPath)
+		if err != nil {
+			// has no target asset to be overried, skip
+			return
+		}
+		defer targetFile.Close()
+
+		sourceFile, err := os.Open(path.Join(distroResDir, assetName))
+		if err != nil {
+			log.Fatal("Failed to open source file", zap.String("path", assetName), zap.Error(err))
```

Comment:
I don't think it is a good behavior to panic everywhere in your logic. previous `rewrite` function are purely based on *generated embedded assets*, which has an *expected content* and *should never miss any files*. So that it is fine to panic when there are vfs IO issues because they *are unexpected programming errors*. However in distro I don't think this still stands. For example, failed to read the dynamic distro file should be tolerated, as it is a file exists on user's FS and it is very possible that the read is failed (e.g. wrong permissions).

Reply:
OK. I thought they should crash as early as possible if the distribution config is not correct, so they won't get the wrong brand names/logos.